tiny_cnn/layers/average_pooling_layer.h

```diff
@@ -147,12 +226,12 @@ class average_pooling_layer : public partial_connected_layer<Activation> {
         cnn_size_t dxmax = std::min(pooling_size, in_.width_ - x);
         cnn_size_t dstx = x / stride_;
         cnn_size_t dsty = y / stride_;
-
+        cnn_size_t outidx = out_.get_index(dstx, dsty, inc);
         for (cnn_size_t dy = 0; dy < dymax; ++dy) {
             for (cnn_size_t dx = 0; dx < dxmax; ++dx) {
                 this->connect_weight(
                     in_.get_index(x + dx, y + dy, inc),
-                    out_.get_index(dstx, dsty, inc),
+                    outidx,
                     inc);
             }
         }
```

Comment:
well, what I suggest is to have implemented `void tiny_average_pooling_kernel ( /* whatever needed params */ )` (for forward and back). The current backend API is a bit rought in this sense. Next refactor will include all kernels functions as templated classes/functors where the template probably will be the backed type itself.